tests/core/service/health/health_service_test.py

```diff
@@ -0,0 +1,90 @@
+from unittest.mock import MagicMock, AsyncMock
+import pytest
+
+from symphony.bdk.gen import ApiException
+from symphony.bdk.gen.agent_model.agent_info import AgentInfo
+from tests.utils.resource_utils import object_from_json_relative_path
+
+from symphony.bdk.core.service.health.health_service import HealthService
+
+from symphony.bdk.gen.agent_api.system_api import SystemApi
+from symphony.bdk.gen.agent_api.signals_api import SignalsApi
+from symphony.bdk.gen.agent_model.v3_health import V3Health
+
+
+@pytest.fixture()
+def mocked_system_api_client():
+    api_client = MagicMock(SystemApi)
+    api_client.v3_health = AsyncMock()
+    api_client.v3_extended_health = AsyncMock()
+    return api_client
+
+
+@pytest.fixture()
+def mocked_signals_api_client():
+    api_client = MagicMock(SignalsApi)
+    api_client.v1_info_get = AsyncMock()
+    return api_client
+
+
+@pytest.fixture()
+def health_service(mocked_system_api_client, mocked_signals_api_client):
+    return HealthService(mocked_system_api_client, mocked_signals_api_client)
+
+
+@pytest.mark.asyncio
+async def test_health_check(health_service, mocked_system_api_client):
+    mocked_system_api_client.v3_health.return_value = object_from_json_relative_path(
+        "health_response/health_check.json")
+
+    health: V3Health = await health_service.health_check()
+
+    assert health.status == "UP"
+
+
+@pytest.mark.asyncio
+async def test_health_check_failed(health_service, mocked_system_api_client):
+    mocked_system_api_client.v3_health.side_effect = ApiException(400)
+
+    with pytest.raises(ApiException):
+        await health_service.health_check()
+
+
+@pytest.mark.asyncio
+async def test_health_check_extended(health_service, mocked_system_api_client):
+    mocked_system_api_client.v3_extended_health.return_value = object_from_json_relative_path(
+        "health_response/health_check_extended.json")
+    health: V3Health = await health_service.health_check_extended()
+
+    assert health.status == "UP"
+    assert health.services.pod.version == "1.57.0"
+    assert health.services.datafeed.status == "UP"
+    assert health.users.agentservice.status == "UP"
+
+
+@pytest.mark.asyncio
+async def test_health_check_extended_failed(health_service, mocked_system_api_client):
+    mocked_system_api_client.v3_extended_health.side_effect = ApiException(400)
+
+    with pytest.raises(ApiException):
+        await health_service.health_check_extended()
```

Comment:
A question that have been asked on Java BDK side: do we really want to return an exception when HealthCheck is down? Isn't something that we want to handle at service level and just return `health.status == "DOWN"`?

Reply:
current behavior => raises an exception if the status is "DOWN"
From a Developer experience POV, as per our discussion, we will wait for users' feedback to change the current behavior to return "DOWN" instead of raising an exception.

As far as I am concerned:
- If the agent is DOWN, we can't have a pod status.
- Even if the agent is DOWN we can still perform calls to the POD for other actions.
From a functional POV knowing that the agent is down and the pod is not is important (maybe that should feature request)